library/Icingadb/Model/Behavior/HasProblematicParent.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/* Icinga DB Web | (c) 2024 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Model\Behavior;
+
+use Icinga\Module\Icingadb\Model\DependencyEdge;
+use ipl\Orm\AliasedExpression;
+use ipl\Orm\ColumnDefinition;
+use ipl\Orm\Exception\InvalidColumnException;
+use ipl\Orm\Query;
+use ipl\Sql\Expression;
+use ipl\Stdlib\Filter;
+use ipl\Orm\Contract\RewriteColumnBehavior;
+use ipl\Orm\Contract\QueryAwareBehavior;
+
+/**
+ * Behavior to check if the service has a problematic parent
+ */
+class HasProblematicParent implements RewriteColumnBehavior, QueryAwareBehavior
+{
+    /** @var Query */
+    protected $query;
+
+    public function setQuery(Query $query): self
+    {
+        $this->query = $query;
+
+        return $this;
+    }
+
+    public function rewriteColumn($column, ?string $relation = null): ?AliasedExpression
+    {
+        if (! $this->isSelectableColumn($column)) {
+            return null;
+        }
+
+        $path = 'from.';
```

Comment:
As discussed offline, instead of using the `Query::createSubQuery` method to form the basic sub query and then join the relevant tables. The entire sub query is manually created. Hence, this line of code becomes obsolete.